_aggregations/pipeline/moving_fn.md

```diff
@@ -0,0 +1,611 @@
+{
+  "size": 0,
+  "aggs": {
+    "my_date_histo": {
+      "date_histogram": {
+        "field": "order_date",
+        "calendar_interval": "week"
+      },
+      "aggs": {
+        "the_sum": {
+          "sum": { "field": "taxful_total_price" }
+        },
+        "the_movavg": {
+          "moving_fn": {
+            "buckets_path": "the_sum",
+            "window": 2,
+            "script": "return (values.length < 2 ? Double.NaN : (values[0]>values[1] ? values[0] : values[1]))"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The example returns the results of the calculation starting in bucket three, where enough previous data exists to perform the calculation:
+
+<details open markdown="block">
+  <summary>
+    Response
+  </summary>
+
+```json
+{
+  "took": 7,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 4675,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "my_date_histo": {
+      "buckets": [
+        {
+          "key_as_string": "2025-03-24T00:00:00.000Z",
+          "key": 1742774400000,
+          "doc_count": 582,
+          "the_sum": {
+            "value": 41455.5390625
+          },
+          "the_movavg": {
+            "value": null
+          }
+        },
+        {
+          "key_as_string": "2025-03-31T00:00:00.000Z",
+          "key": 1743379200000,
+          "doc_count": 1048,
+          "the_sum": {
+            "value": 79448.60546875
+          },
+          "the_movavg": {
+            "value": null
+          }
+        },
+        {
+          "key_as_string": "2025-04-07T00:00:00.000Z",
+          "key": 1743984000000,
+          "doc_count": 1048,
+          "the_sum": {
+            "value": 78208.4296875
+          },
+          "the_movavg": {
+            "value": 79448.60546875
+          }
+        },
+        {
+          "key_as_string": "2025-04-14T00:00:00.000Z",
+          "key": 1744588800000,
+          "doc_count": 1073,
+          "the_sum": {
+            "value": 81277.296875
+          },
+          "the_movavg": {
+            "value": 79448.60546875
+          }
+        },
+        {
+          "key_as_string": "2025-04-21T00:00:00.000Z",
+          "key": 1745193600000,
+          "doc_count": 924,
+          "the_sum": {
+            "value": 70494.2578125
+          },
+          "the_movavg": {
+            "value": 81277.296875
+          }
+        }
+      ]
+    }
+  }
+}
+```
+</details>
+
+
+## Example: Moving average
+
+The `moving_fn` aggregation replaces the deprecated `moving_avg` aggregation. The `moving_fn` aggregation is similar to the `moving_avg` aggregation but is more versatile since it computes arbitrary functions, not only averages. All of the predefined `moving_avg` functions are implemented in `moving_fn` as well. 
```

Comment:
```suggestion
The `moving_fn` aggregation replaces the deprecated `moving_avg` aggregation. The `moving_fn` aggregation is similar to the `moving_avg` aggregation but is more versatile since it computes arbitrary functions instead of only averages. All of the predefined `moving_avg` functions are implemented in `moving_fn` as well. 
```